cert-provider/cert-provider.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+
+ACME_BIN="$(realpath ~/.acme.sh/acme.sh)"
+
+# colour output helpers...
+reset=$(tput -T xterm sgr0)
+red=$(tput -T xterm setaf 1)
+green=$(tput -T xterm setaf 2)
+yellow=$(tput -T xterm setaf 3)
+blue=$(tput -T xterm setaf 4)
+
+function logError() {
+    echo "${red}[Error]${reset} $1"
+}
+
+function logWarn() {
+    echo "${yellow}[Warn]${reset} $1"
+}
+
+function logInfo() {
+    echo "${blue}[Info]${reset} $1"
+}
+
+function logSuccess() {
+    echo "${green}[Success]${reset} $1"
+}
+
+# join domains with -d separator
+join_domains()  { echo "$1" | sed -e "s/,/ -d /g" -e "s/^/-d /g"; }
```

Comment:
Why mix up two forms of defining a function? I think this form is the generic one, while using the `function` keyword as above, bash-specific (or the other way round). It is good to use only one or the other (preferably the generic one, I think that's what we use elsewhere in scripts).

Reply:
dropped in favour of @wrboyce suggestion :+1: